src/v/cloud_storage/remote_segment.cc

```diff
@@ -126,63 +133,113 @@ const model::term_id remote_segment::get_term() const {
 ss::future<> remote_segment::stop() {
     vlog(_ctxlog.debug, "remote segment stop");
     _as.request_abort();
+    _bg_cvar.broken();
     return _gate.close();
 }
 
 ss::future<ss::input_stream<char>>
-remote_segment::data_stream(size_t pos, ss::io_priority_class io_priority) {
+remote_segment::data_stream(size_t pos, ss::io_priority_class) {
     vlog(_ctxlog.debug, "remote segment file input stream at {}", pos);
     ss::gate::holder g(_gate);
     // Hydrate segment on disk
     auto full_path = co_await hydrate();
     // Create a file stream
-    auto opt = co_await _cache.get(full_path, pos, io_priority);
+    auto opt = co_await _cache.get(full_path, pos);
     if (opt) {
         co_return std::move(opt->body);
     }
     throw remote_segment_exception(
-      fmt::format("Segment {} is not in the cache", full_path));
+      fmt::format("Segment {} already evicted", full_path));
 }
 
-ss::future<std::filesystem::path> remote_segment::hydrate() {
-    ss::gate::holder g(_gate);
+ss::future<> remote_segment::run_hydrate_bg() {
+    ss::gate::holder guard(_gate);
     auto full_path = _manifest.get_remote_segment_path(_path);
-    vlog(_ctxlog.debug, "hydrating segment {}", full_path);
-    auto poll_deadline = ss::lowres_clock::now() + cache_poll_timeout;
-    while (co_await _cache.is_cached(full_path)
-           == cache_element_status::in_progress) {
-        vlog(
-          _ctxlog.debug,
-          "segment {} is being written to cache, waiting...",
-          full_path);
-        // poll cache periodically until status changes
-        co_await ss::sleep_abortable(cache_poll_interval, _as);
-        if (ss::lowres_clock::now() > poll_deadline) {
-            break;
+    try {
+        while (!_gate.is_closed() && !_as.abort_requested()) {
```

Comment:
Trying to understand if we really need `_as` at all. Looks like `stop()` will close the gate and break the cvar, so if we waited on cvar, we will exit via exception and if we were in the loop body we will exit on the next iter because the gate will be closed. So, does the abort source really add anything?